### PR TITLE
feat: Quick account switcher

### PR DESCRIPTION
**Pull request by 0xCmdrKeen**
Since #715 has blocked and inactive for a while, and I really miss this feature from Apollo, I thought I'd try my own version of this.

This makes tapping the Profile button in the tab bar pull up the account switcher IF the profile page is already active in the profile tab (and hasn't been scrolled down on).

This means tapping the Profile button TWICE will bring up the account switcher also.

If you open the Profile page by tapping the button once, and then navigate away by clicking one of the links on it, tapping the profile button again will bring you back to the profile page, as before.

https://github.com/aeharding/voyager/assets/98132670/1f41e4db-5917-44bb-ac38-1c94948861b5

### CODE REVIEW

File: src/features/auth/PageContext.tsx [hunk 1]
```diff
@@ -59,6 +62,13 @@ interface IPageContext {
     comment?: CommentView,
     comments?: CommentView[],
   ) => void;
+
+  presentAccountSwitcher: (
```
Comment (aeharding) [inline src/features/auth/PageContext.tsx:66]: Let's have this function take no arguments, and then add a function in PageContext to call presentAccountSwitcher with cssClass = small.

File: src/TabBar.tsx [hunk 2]
```diff
@@ -136,6 +138,10 @@ const TabBar: CustomTabBarType = forwardRef(function TabBar(props, ref) {
 
     if (scrollUpIfNeeded(activePageRef?.current)) return;
 
+    // if the profile page is already open, show the account switcher
+    if (router.getRouteInfo()?.pathname === "/profile")
```
Comment (aeharding) [inline src/TabBar.tsx:142]: `location.pathname` instead to be consistent with the rest of the component :)

**Review by aeharding — COMMENTED** [review]
Looks good, I like your idea!

File: src/features/auth/PageContext.tsx (hunk 1 above)
Comment (0xCmdrKeen) [reply src/features/auth/PageContext.tsx:66]: Sounds good

File: src/TabBar.tsx (hunk 2 above)
Comment (0xCmdrKeen) [reply src/TabBar.tsx:142]: No problem

**Comment by aeharding** [comment]
Looks good! I had one problem where the presentingElement of the login modal from the account switcher wasn't properly attaching so I pushed a fix for that.

**Pull request merged**